src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2968,6 +2968,18 @@ def set_up_network_profile(self, mc: ManagedCluster) -> ManagedCluster:
         mc = super().set_up_network_profile(mc)
         network_profile = mc.network_profile
 
+        # network_plugin is typically defaulted to kubenet. AKS-RP is moving
+        # away from specifying this default in the API and making it based
+        # on the k8s version being used. The CLI should not be responsible
+        # for setting default values and should pass properties as empty
+        # unless specified by the user.
+        if (
+            network_profile.network_plugin is not None and
+            self.context.raw_param.get("network_plugin") is None and
+            self.context.decorator_mode == DecoratorMode.CREATE
```

Comment:
may omit `self.context.decorator_mode == DecoratorMode.CREATE` as `set_up_network_profile` is only called in the code path of `aks create` (get_xxx could be used by both create and update and DecoratorMode is used there)

Reply:
fixed